File: server/services/getCart.go
package services
import (
	"encoding/json"
	"net/http"
	"tashvik-oms/server/database"
	"tashvik-oms/server/models"
	// "fmt"
)

func GetCart(res http.ResponseWriter , request *http.Request){
	var cart models.Cart;
	userID := request.URL.Query().Get("id")
	if userID == "" {
		http.Error(res, "Missing user ID", http.StatusBadRequest)
		return
	}
	result := database.DB.Where("user_id = ? AND status = ?", userID, 1).Find(&cart)
	if result.Error != nil {
		http.Error(res, "Error fetching cart", http.StatusInternalServerError)
		return
	}
	res.Header().Set("Content-Type", "application/json")
	json.NewEncoder(res).Encode(cart)
}

File: public/src/apis/getCart.ts
export const getCart = async () => {
    try {
        const response = await fetch('https://merry-normally-panda.ngrok-free.app/GetCart?id=1');
        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }
        const data = await response.json();
        // console.log(data)
        return data;
    } catch (error) {
        console.error('Error fetching cart:', error);
        throw error;
    }
  }